src/Turtle/Prelude.hs

```diff
@@ -1331,6 +1332,28 @@ fork io = using (managed (withAsync io))
 wait :: MonadIO io => Async a -> io a
 wait a = liftIO (Control.Concurrent.Async.wait a)
 
+{-| Run a list of actions in parallel using fork and wait
+
+>>> sh . using $ parallel $ fmap (\i -> Turtle.shells "sleep 2 && date" empty) [1..10]
+Sat Nov 19 17:26:52 CET 2016
+Sat Nov 19 17:26:52 CET 2016
+Sat Nov 19 17:26:52 CET 2016
+Sat Nov 19 17:26:52 CET 2016
+Sat Nov 19 17:26:52 CET 2016
+Sat Nov 19 17:26:52 CET 2016
+Sat Nov 19 17:26:52 CET 2016
+Sat Nov 19 17:26:52 CET 2016
+Sat Nov 19 17:26:52 CET 2016
+Sat Nov 19 17:26:52 CET 2016
+
+-}
+parallel :: MonadManaged managed => [IO ()] -> managed ()
```

Comment:
You could actually probably simplify the type to:

```haskell
parallel :: [IO ()] -> IO ()
```

... by just wrapping the whole thing in `runManaged` since you don't really benefit from keeping the result inside of `Managed`.  However, then this wouldn't be really `turtle`-specific at all and I'm pretty sure there's a library on Hackage that already provides such a function

Also, I think for this to be more in the spirit of the GNU `parallel` command you'd want to be able to collect the results so they could be processed further, using a type like this:

```haskell
-- | Fork all the `IO` actions and stream their results as they complete, preserving order
parallel :: [IO a] -> Shell a
```

... and then if you want to ignore their results, you would just wrap the whole thing in `sh`.  You could implement it something like this:

```haskell
parallel :: [IO a] -> Shell a
parallel ios = do
    asyncs <- mapM fork ios
    async  <- select asyncs
    wait async
```

So, for example, this code:

```haskell
main = view (parallel (map sleep [1, 2, 3, 1, 2, 1]))
```

... would print:

```haskell
() -- After 1 second
() -- After 2 seconds
() -- After 3 seconds
() -- After 3 seconds
() -- After 3 seconds
() -- After 3 seconds
```